iotilecore/iotile/core/hw/reports/signed_list_format.py

```diff
@@ -23,6 +29,41 @@ class SignedListReport(IOTileReport):
     def __init__(self, rawreport, **kwargs):
         super(SignedListReport, self).__init__(rawreport, signed=True, encrypted=False, **kwargs)
 
+    @classmethod
+    def KeyTypeToStreamType(cls, key_type):
+        """Converts key type into the type of signed report that can be encrypted with this key
+
+        Args:
+            int: a type of the key, see AuthProvider
+
+        Returns:
+            ReportSignatureFlags: a type of the stream report
+        """
+        if key_type == AuthProvider.NoKey:
+            return ReportSignatureFlags.SIGNED_WITH_HASH
+        elif key_type == AuthProvider.UserKey:
+            return ReportSignatureFlags.SIGNED_WITH_USER_KEY
+        elif key_type == AuthProvider.DeviceKey:
+            return ReportSignatureFlags.SIGNED_WITH_DEVICE_KEY
+
+    @classmethod
+    def StreamTypeToKeyType(cls, stream_type):
+        """Converts the stream type into the key type that was used to encryped this stream report
+
+        Args:
+            ReportSignatureFlags: a type of the stream report
+
+        Returns:
+            int: type of key, see AuthProvider
+        """
+
+        if stream_type in [ReportSignatureFlags.SIGNED_WITH_HASH]:
```

Comment:
This should be `==` as well since there's a single option.

Reply:
Got it